mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MesosSchedulerClient.java

```diff
@@ -97,35 +99,15 @@ public MesosSchedulerClient(
             .pipelineConfigurator(new HttpClientPipelineConfigurator<>())
             .build();
 
-        createPost = (Send s) -> {
-            final byte[] bytes = sendCodec.encode(s);
-            HttpClientRequest<ByteBuf> request = HttpClientRequest.createPost(mesosUri.getPath())
-                .withHeader("User-Agent", userAgent.toString())
-                .withHeader("Content-Type", sendCodec.mediaType())
-                .withHeader("Accept", receiveCodec.mediaType())
-                .withHeader("Accept", receiveCodec.mediaType());
-
-            final String userInfo = mesosUri.getUserInfo();
-            if (userInfo != null) {
-                //Won't actually work until https://issues.apache.org/jira/browse/MESOS-3923 is fixed
-                request = request.withHeader(
-                    HttpHeaders.AUTHORIZATION,
-                    String.format("Basic %s", Base64.getEncoder().encodeToString(userInfo.getBytes()))
-                );
-            }
-            return just(
-                request
-                    .withContent(bytes)
-            );
-        };
+        createPost = curryCreatePost(mesosUri, sendCodec, receiveCodec, userAgent);
     }
 
     @NotNull
-    public Observable<Receive> openEventStream(@NotNull final Send subscription) {
-        return createPost.call(subscription)
+    public AwaitableSubscription openStream() {
+        final Observable<Receive> receives = createPost.call(subscribe)
```

Comment:
Was there a reason for moving the subscribe call from a method parameter to a constructor parameter?